Fix for distorted Thumbnails

KevinHansen23 opened the pull request:
### 1. Why is this change necessary?
Thumbnails are distorted if the config for thumbnail width and height isn't equal

### 2. What does this change do, exactly?
To check whether the KeepAspectRatio config is set **or** the configured width and height is different doesn't make sense.
Currently every Thumbnail that has a different width and height config is distorted.
You only have to check, if the KeepAspectRatio config is set and then calculate the correct thumbnail size based on a factor and not the aspect ratio.

Here is a small table that represents the new calculation:

Image width | Image height | Factor | Thumbnail width | Thumbnail height | New width | New height
-- | -- | -- | -- | -- | -- | --
2000 | 1000 | 0,4 | 800 | 600 | 800 | 400
2000 | 1000 | 0,3 | 600 | 800 | 600 | 300
2000 | 1000 | 0,4 | 800 | 800 | 800 | 400
1000 | 2000 | 0,3 | 800 | 600 | 300 | 600
1000 | 2000 | 0,4 | 600 | 800 | 400 | 800
1000 | 2000 | 0,4 | 800 | 800 | 400 | 800

### 3. Describe each step to reproduce the issue or behaviour.
Create a new Thumbnail config width deviating width and height -> upload images.

KevinHansen23 committed:
Rename {2023-08-04}-Fix-for-distorted-Thumbnails.md to 2023-08-04-Fix-for-distorted-Thumbnails.md

Create {2023-08-04}-Fix-for-distorted-Thumbnails.md

Added Changelog

Update ThumbnailService.php

Fix for distorted Thumbnails

Update ThumbnailService.php

Fix for distorted Thumbnails

Create {2023-08-04}-Fix-for-distorted-Thumbnails.md

Added Changelog

Rename {2023-08-04}-Fix-for-distorted-Thumbnails.md to 2023-08-04-Fix-for-distorted-Thumbnails.md

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/shopware/shopware?pullRequest=3253) <br/>All committers have signed the CLA.

KevinHansen23 commented:
@shyim Do you have an Update on this for me?

leduc92 commented:
Hi @KevinHansen23 , we're pulling the MR into our internal repo and processing the next step.

ghost commented:
Hello,

thank you for creating this pull request.
I have opened an issue on our Issue Tracker for you. See the issue link: https://issues.shopware.com/issues/NEXT-30672

Please use this issue to track the state of your pull request.

leduc92 commented:
Hi @KevinHansen23 , sorry for the late response. We were working on pulling your merge request to our internal repository, but we saw some needs. Would you mind to update your merge request to:
- Add the changelog file.
- Squash your commit.
- It would be nice if you could add the unitest to cover your changes.

Once again, thanks for your willingness and contribution.

KevinHansen23 commented:
Hi @leduc92,

I've added the changelog. 

Regarding squashing the commits: We don't work with Github and I can't find an Option to do this in the web Version.
Regarding the Test: The ThumbnailService Class does not have any tests. And the Method I changed is private.

celha commented:
@KevinHansen23 I'll take this over! 
But before we can merge this, you need to squash the commits as @leduc92 said. :) 
You can find an example doc here: https://www.git-tower.com/learn/git/faq/git-squash

celha commented:
@KevinHansen23 I think your rebase didn't go through. You still have 3 commits. The goal here is only to have one left :) Please rebase again in that way so that I can get your pr merged.

ghost commented:
Hello,

thank you for creating this pull request.
I have opened an issue on our Issue Tracker for you. See the issue link: https://issues.shopware.com/issues/NEXT-30672

Please use this issue to track the state of your pull request.

KevinHansen23 commented:
@celha should be okay now.

marcelbrode commented:
Merged! Thanks for your contribution! :blue_heart: :tada:
